server/src/main/java/org/cloudfoundry/identity/uaa/authentication/manager/PasswordGrantAuthenticationManager.java

```diff
@@ -175,8 +177,11 @@ Authentication oidcPasswordGrant(Authentication authentication, OIDCIdentityProv
         MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
 
         if (ClientAuthentication.PRIVATE_KEY_JWT.equals(calcAuthMethod)) {
+            /* ensure that the dynamic lookup of the cert and/or key for private key JWT works for an alias IdP in a
+             * custom IdZ */
+            final boolean allowDynamicValueLookupInCustomZone = hasText(identityProvider.getAliasZid()) && hasText(identityProvider.getAliasId());
```

Comment:
I intentionally did not add the `aliasEntitiesEnabled` check here. Whenever an IdP with alias (that also references a dynamic value) exists and `aliasEntitiesEnabled` is deactivated afterwards, the check would lead to failing logins.

In general, the `aliasEntitiesEnabled` flag should only prevent new aliases to be created and existing ones to be updated. It should not block entities with alias from working in, e.g., login flows.